cmd/stat.go

```diff
@@ -0,0 +1,222 @@
+	Key               string            `json:"name"`
+	Date              time.Time         `json:"lastModified"`
+	Size              int64             `json:"size"`
+	ETag              string            `json:"etag"`
+	Type              string            `json:"type"`
+	EncryptionHeaders map[string]string `json:"encryption"`
+	Metadata          map[string]string `json:"metadata"`
+}
+
+// String colorized string message.
+func printStat(stat statMessage, maxKeyLen int) {
+	// Format properly for alignment based on maxKey length
+	stat.Key = fmt.Sprintf("%-10s: %-*.*s", "Name", maxKeyLen, maxKeyLen, stat.Key)
+	console.Println(console.Colorize("Name", stat.Key))
+	console.Println(fmt.Sprintf("%-10s: %s ", "Date", stat.Date.Format(printDate)))
+	console.Println(fmt.Sprintf("%-10s: %-6s ", "Size", humanize.IBytes(uint64(stat.Size))))
+	if stat.ETag != "" {
+		console.Println(fmt.Sprintf("%-10s: %s ", "ETag", stat.ETag))
+	}
+	console.Println(fmt.Sprintf("%-10s: %s ", "Type", stat.Type))
+
+	var maxKey = 0
+	for k := range stat.Metadata {
+		if len(k) > maxKey {
+			maxKey = len(k)
+		}
+	}
+	if len(stat.Metadata) > 0 {
+		console.Println(fmt.Sprintf("%-10s:", "Metadata"))
+		for k, v := range stat.Metadata {
+			console.Println(fmt.Sprintf("  %-*.*s: %s ", maxKey, maxKey, k, v))
+		}
+	}
+	maxKey = 0
+	for k := range stat.EncryptionHeaders {
+		if len(k) > maxKey {
+			maxKey = len(k)
+		}
+	}
+	if len(stat.EncryptionHeaders) > 0 {
+		console.Println(fmt.Sprintf("%-10s:", "Encrypted"))
+		for k, v := range stat.EncryptionHeaders {
+			console.Println(fmt.Sprintf("  %-*.*s: %s ", maxKey, maxKey, k, v))
+		}
+	}
+	console.Println()
+}
+
+// Prints all the stats.
+func printStats(stats ...statMessage) {
+	var maxKey = 0
+	for _, stat := range stats {
+		if len(stat.Key) > maxKey {
+			maxKey = len(stat.Key)
+		}
+	}
+
+	for _, stat := range stats {
+		if !globalJSON {
+			printStat(stat, maxKey)
+		} else {
+			console.Println(stat.JSON())
+		}
+	}
+}
+
+// JSON jsonified content message.
+func (c statMessage) JSON() string {
+	c.Status = "success"
+	jsonMessageBytes, e := json.Marshal(c)
+	fatalIf(probe.NewError(e), "Unable to marshal into JSON.")
+
+	return string(jsonMessageBytes)
+}
+
+// parseContent parse client Content container into printer struct.
+func parseStat(c *clientContent) statMessage {
+	content := statMessage{}
+	content.Date = c.Time.Local()
+	// guess file type.
+	content.Type = func() string {
+		if c.Type.IsDir() {
+			return "folder"
+		}
+		return "file"
+	}()
+	content.Size = c.Size
+	content.Key = getKey(c)
+	content.Metadata = c.Metadata
+	content.ETag = c.ETag
+	content.EncryptionHeaders = c.EncryptionHeaders
+	return content
+}
+func getKey(c *clientContent) string {
+	switch {
+	// for windows make sure to print in 'windows' specific style.
+	case runtime.GOOS == "windows":
+		c.URL.Path = strings.Replace(c.URL.Path, "/", "\\", -1)
+		c.URL.Path = strings.TrimSuffix(c.URL.Path, "\\")
```

Comment:
Also a repeated code, we can have a common translation function for both ls and stat here.

Reply:
done.